packages/web-react/src/components/Dialog/Dialog.tsx

```diff
@@ -14,7 +14,7 @@ const Dialog = (props: DialogProps, ref: ForwardedRef<HTMLDialogElement | null>)
   // handles toggling based on state
   const { closeDialog } = useDialog(dialogElementRef as MutableRefObject<HTMLDialogElement | null>, isOpen);
   const handleClickOutside = (event: Event) => {
-    if (closeOnBackdropClick) {
+    if (closeOnBackdropClick && event.target === dialogElementRef.current) {
```

Comment:
Please add a comment to this condition in the same way I did in the `useClickOutside` hook. I think this is an important place and condition that deserves this type of care ;-) Thanks.